### ISSUE TITLE
Cannot install/find OpenSSL on OSX 

### ISSUE THREAD

**Issue by marimeireles** [nextcloud/desktop#1538]
### Expected behaviour
CMake should find the path to my openssl instalation.

### Actual behaviour
```
Could NOT find LIBCLOUDPROVIDERS (missing: LIBCLOUDPROVIDERS_INCLUDE_DIR LIBCLOUDPROVIDERS_LIBRARY)
CMake Error
```

Unfortunately I don't have the rest of the log anymore, closed my terminal and lost it :(
But I can try to reproduce in a different Mac if this is a relevant issue. I use:
* MacOS Mojave
* Qt5.1.13 from brew
* openssl 1.1 from brew

### Steps to reproduce
1. clone the repo
2. create the dirs specified on the readme
3. try to run `cmake .. -DCMAKE_INSTALL_PREFIX=~/nextcloud-desktop-client -DCMAKE_BUILD_TYPE=Debug -DNO_SHIBBOLETH=1`

**Comment by marimeireles** [comment]
Also as an additional feedback for Mac users I got stuck in this step for a while:

```
  Could NOT find Qt5Keychain (missing: QTKEYCHAIN_LIBRARY
  QTKEYCHAIN_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.15.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindQt5Keychain.cmake:47 (find_package_handle_standard_args)
  src/CMakeLists.txt:13 (find_package)
```

Because I didn't know what was the right path to set the `QTKEYCHAIN_LIBRARY` since the qtkeychain repo doesn't offer any guiding on that matter. For me it was `export QTKEYCHAIN_LIBRARY="/usr/local/lib/libqt5keychain.dylib"`
Maybe it's worth it to include in the docs/wiki? If so, let me know, I can do it :)

**Comment by camilasan** [comment]
This @marimeireles should fix your problem:
-DOPENSSL_ROOT_DIR=/c/OpenSSL/bin/ -DOPENSSL_INCLUDE_DIR=/c/OpenSSL/include/ -DOPENSSL_LIBRARIES=/c/OpenSSL/lib/

/c/OpenSSL/ -> replace for wherever the bin/include/lib folder of the openssl library is.

**Comment by camilasan** [comment]
> Maybe it's worth it to include in the docs/wiki? If so, let me know, I can do it :)

Yes!

**Comment by elsiehupp** [comment]
Hi! I think this is related to #2893 and may be fixed with #2931.

@marimeireles the OpenSSL problem in particular is because macOS comes preinstalled with LibreSSL, and if you install OpenSSL via Homebrew it doesn't link it higher on the $PATH in order to avoid conflicts. My workaround was to link OPENSSL_ROOT_DIR to the Homebrew OpenSSL install in the `zsh` profile with the following command:
```
% echo 'export OPENSSL_ROOT_DIR="/usr/local/opt/openssl@1.1"' >> ~/.zshrc
```
